props/saber_fett263_G_buttons.h

```diff
@@ -0,0 +1,445 @@
+    if (!battle_mode_) {
+       battle_mode_ = true;
+       hybrid_font.SB_Force();
+    } else {
+       battle_mode_ = false;
+       hybrid_font.SB_Force();
+    }
+    return true;
+
+// Auto Lockup Mode
+case EVENTID(BUTTON_NONE, EVENT_CLASH, MODE_ON):
+       if (battle_mode_) {
+       clash_swing_ = millis();
+       swing_blast_ = false;
+       if(clash_swing_ > last_swing_) {
+           SaberBase::SetLockup(SaberBase::LOCKUP_NORMAL);
+           swing_blast_ = false;
+           auto_lockup_on_ = true;
+           SaberBase::DoBeginLockup();
+           return true;
+       }
+      } else {
+           return false;
+	 }
+
+case EVENTID(BUTTON_NONE, EVENT_STAB, MODE_ON):
+       if (battle_mode_) {
+       clash_swing_ = millis();
+       swing_blast_ = false;
+       if(clash_swing_ > last_swing_) {
+          if ((fusor.angle1() + M_PI / 2) * (32768 / M_PI) < 8000) {
+            SaberBase::SetLockup(SaberBase::LOCKUP_DRAG);
+          } else {
+            SaberBase::SetLockup(SaberBase::LOCKUP_MELT);
+          }
+           swing_blast_ = false;
+           auto_melt_on_ = true;
+           SaberBase::DoBeginLockup();
+           return true;
+       }
+           } else {
+           return false;
+           }
+	
+
+case EVENTID(BUTTON_NONE, EVENT_SWING, MODE_OFF):  
+	  if(millis() > 3000) { 
+          On();
+          battle_mode_ = true;
+        }
+        return true;
+
+case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_ON):
+        if (millis() > last_twist_ + 3000) {
+        Off();
+        last_twist_ = millis();
+        saber_off_time_ = millis();
+	   }
+        return true;
+
+// Optional Gestures use defines to enable
+
+#ifdef FETT263_STAB_ON
+case EVENTID(BUTTON_NONE, EVENT_STAB, MODE_OFF):
+        On();
+        return true;
+#endif
+
+#ifdef FETT263_TWIST_ON
+case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_OFF):
+        if (millis() > last_twist_ + 3000) {
+        On();
+        last_twist_ = millis();
+        }
+        return true;
+#endif
+
+#ifdef FETT263_MULTI_PHASE
+case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_ON | BUTTON_AUX):
+	   next_preset();
+        return true;
+
+case EVENTID(BUTTON_NONE, EVENT_TWIST, MODE_ON | BUTTON_POWER):
+        previous_preset();
+        return true;
+
+#endif
+        
+        // Events that needs to be handled regardless of what other buttons
+        // are pressed.
+      case EVENTID(BUTTON_POWER, EVENT_RELEASED, MODE_ANY_BUTTON | MODE_ON):
+      case EVENTID(BUTTON_AUX, EVENT_RELEASED, MODE_ANY_BUTTON | MODE_ON):
+        if (SaberBase::Lockup()) {
+          SaberBase::DoEndLockup();
+          SaberBase::SetLockup(SaberBase::LOCKUP_NONE);
+          return true;
+        }
+    }
+    return false;
+  }
+private:
+  bool pointing_down_ = false;
+  bool swing_blast_ = false;
+  bool mode_volume_ = false;
+  bool auto_lockup_on_ = false;
+  bool auto_melt_on_ = false;
+  bool battle_mode_ = false;
+  uint32_t clash_swing_ = millis();
```

Comment:
clash swing?
What does that mean?
Can we come up with a better name?

Reply:
I renamed to clash_impact_, it's used to give a delay before checking next gesture

Reply:
clash impact isn't a whole lot better, because it doesn't give any clue as to what the unit of the variable is.
Maybe last_clash_impact_  or clash_impact_millis_?